File: problems/greedy/TICT3_1_minimum_number_of_coins/solution.ts
function minimumCoins(amount: number) {
  let count = 0
  const coinTypes = [500,100,50,10]

  for (const coin of coinTypes) {
    count += Math.floor(amount / coin)
    amount %= coin
  }
  return count
}

function TCCT3_1Main() {
  const amount = 1260
  console.log(minimumCoins(amount))
}

TCCT3_1Main()

File: problems/greedy/TICT3_1_minimum_number_of_coins/solution.go
package main

import (
	"fmt"
)

func minimumCoins(amount int) int {
	count := 0
	coinTypes := []int{500, 100, 50, 10}
	for _, coin := range coinTypes {
		count += amount / coin
		amount %= coin
	}

	return count
}

func main() {
	amount := 1260
	fmt.Println(minimumCoins(amount))
}
